vue-app/src/api/recipient-registry-optimistic.ts

```diff
@@ -152,7 +155,7 @@ export async function getRequests(
     requestResolvedFilter,
     0
   )
-  const requests: Request[] = []
+  const requests: Record<string, Request> = {}
```

Comment:
`getRequests` was returning more than one request per recipient. This is not possible at the contract level. A new request is basically going to overwrite the previous one `requests[recipientId] = Request(...)`.
The issue arises due that it is looping through the `RequestSubmitted` events where you can certainly have more than one submission event per recipient.
This fix is implementing a dictionary where we keep only the last event emitted in it.